ui/de-lib/src/main/java/org/iplantc/de/client/services/impl/MessageServiceFacadeImpl.java

```diff
@@ -141,9 +142,13 @@ public void getMessageCounts(final AsyncCallback<Counts> callback) {
     }
 
     @Override
-    public void deleteAll(AsyncCallback<String> callback) {
+    public void deleteAll(NotificationCategory category, AsyncCallback<String> callback) {
         String address = deProperties.getMuleServiceBaseUrl() + "notifications/delete-all"; //$NON-NLS-1$
 
+        if (!category.equals(NotificationCategory.ALL)) {
```

Comment:
I would invert this `equals` call, just in case `category` is `null`, then an NPE won't get thrown.
Although with an enum you could just use `category != NotificationCategory.ALL`, which is null-safe as well.